tt_metal/impl/debug/dprint_server.cpp

```diff
@@ -581,25 +607,31 @@ void DebugPrintServerContext::PollPrintData(uint32_t hart_mask) {
 
         // Make a copy of the device->core map, so that it can be modified while polling.
         std::map<Device*, vector<CoreCoord>> device_to_core_range_copy;
+        std::map<Device*, set<CoreCoord>> device_to_eth_cores_copy;
         device_to_core_range_lock_.lock();
         device_to_core_range_copy = device_to_core_range_;
+        device_to_eth_cores_copy = device_to_eth_cores_;
         device_to_core_range_lock_.unlock();
 
         // Flag for whether any new print data was found in this round of polling.
         bool new_data_this_iter = false;
         for (auto& device_and_cores : device_to_core_range_copy) {
             chip_id_t chip_id = device_and_cores.first->id();
+            set<CoreCoord>& eth_cores = device_to_eth_cores_copy.at(device_and_cores.first);
             for (auto core: device_and_cores.second) {
-                for (int hart_index = 0; hart_index < DPRINT_NRISCVS; hart_index++) {
+                bool is_eth_core = (eth_cores.count(core) > 0);
```

Comment:
`inline bool is_ethernet_core(const CoreCoord &core, chip_id_t chip_id)` maybe?

Reply:
Added in latest commit